crates/core_arch/src/x86/avx.rs

```diff
@@ -1579,7 +1579,13 @@ pub unsafe fn _mm256_insertf128_si256(a: __m256i, b: __m128i, imm8: i32) -> __m2
 #[rustc_args_required_const(2)]
 #[stable(feature = "simd_x86", since = "1.27.0")]
 pub unsafe fn _mm256_insert_epi8(a: __m256i, i: i8, index: i32) -> __m256i {
-    transmute(simd_insert(a.as_i8x32(), (index as u32) & 31, i))
+    let a = a.as_i8x32();
+    macro_rules! call {
+        ($index:expr) => {
+            simd_insert(a, $index, i)
+        };
+    }
+    transmute(constify_imm5!((index & 31), call))
```

Comment:
There's no need to mask here, it's done for you inside the `constify_imm5` macro. Same with all the other cases.

Reply:
Missed that the `constify_imm*` macros masked the immediate.